test/algorithms/test_statepreparation.cpp

```diff
@@ -70,3 +71,23 @@ TEST_P(StatePreparation, StatePreparationCircuitSimulation) {
     ASSERT_NEAR(expectedAmplitudes[i].imag(), result[i].imag(), EPS);
   }
 }
+
+TEST(StatePreparation, StatePreparationAmplitudesNotNormalized) {
+  const auto amplitudes = std::vector<std::complex<double>>{
+      std::complex<double>{1}, std::complex<double>{1}};
+
+  // cast to avoid linting nodiscard
+  ASSERT_THROW(static_cast<void>(qc::createStatePreparationCircuit(amplitudes)),
+               std::invalid_argument);
+}
+
+TEST(StatePreparation, StatePreparationsAmplitudesNotPowerOf2) {
+  const auto amplitudes =
+      std::vector<std::complex<double>>{std::complex<double>{1 / std::sqrt(3)},
+                                        std::complex<double>{1 / std::sqrt(3)},
+                                        std::complex<double>{1 / std::sqrt(3)}};
+
+  // cast to avoid linting nodiscard
+  ASSERT_THROW(static_cast<void>(qc::createStatePreparationCircuit(amplitudes)),
+               std::invalid_argument);
+}
```

Comment:
```suggestion
  ASSERT_THROW(std::ignore = qc::createStatePreparationCircuit(amplitudes),
               std::invalid_argument);
}

TEST(StatePreparation, StatePreparationsAmplitudesNotPowerOf2) {
  const auto amplitudes =
      std::vector<std::complex<double>>{std::complex<double>{1 / std::sqrt(3)},
                                        std::complex<double>{1 / std::sqrt(3)},
                                        std::complex<double>{1 / std::sqrt(3)}};

  ASSERT_THROW(std::ignore = qc::createStatePreparationCircuit(amplitudes),
               std::invalid_argument);
}
```
There is a slightly more idiomatic way to ignore nodiscard return values.